cirq/devices/noise_model.py

```diff
@@ -58,20 +52,21 @@ def noisy_moments(self, moments: 'Iterable[cirq.Moment]',
             A sequence of OP_TREEs, with the k'th tree corresponding to the
             noisy operations for the k'th moment.
         """
-        if not hasattr(self.noisy_moment, '_not_overridden'):
+        if self._alternative_for('noisy_moments') == 'noisy_moment':
             result = []
             for moment in moments:
                 result.append(self.noisy_moment(moment, system_qubits))
             return result
 
-        if not hasattr(self.noisy_operation, '_not_overridden'):
+        if self._alternative_for('noisy_moments') == 'noisy_operation':
             result = []
             for moment in moments:
                 result.append([self.noisy_operation(op) for op in moment])
             return result
 
         assert False, 'Should be unreachable.'
 
+    @abc_alt.abstractmethod_alternatives('noisy_moments', 'noisy_operation')
```

Comment:
I think this decorator should ideally be like this:

```
@alternative(requires='other_method', implementation=default_implementation)
@alternative(requires='other_method_2', implementation=default_implementation_2)
def noisy_moment(...):
    pass
```

Reply:
I agree this would be cleaner.  The default implementation could then be resolved at class creation.